src/crafting.rs

```diff
@@ -495,23 +528,40 @@ impl Listing {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+enum RecipeSource {
+    Automatic,
+    Discovered,
+    Purchase,
```

Comment:
`Discovered` and `Purchased` are a bit confusing to me when I read them out of context, because it sounds like it refers to the recipe being known. How about `Discoverable` and `Purchasable`?